CASMHMS-5568: EX2500 cabinet doc changes (#2060)
Cray-HPE/docs-csm: operations/node_management/Add_a_Standard_Rack_Node.md

These procedures are intended for trained technicians and support personnel only. Always follow ESD precautions when handling this equipment.

> - An authentication token has been retrieved.

Reviewer: Should this be in a prerequisites section?

Author: Yes it should be

Author: I moved it to be the first step of the procedure.

The example is this procedure adds a User Access Node \(UAN\) or compute node to an HPE Cray standard rack system. This example adds a node to rack number 3000 at U27.

Procedures for updating the Hardware State Manager \(HSM\) or System Layout Service \(SLS\) are similar when adding additional compute nodes or User Application Nodes \(UANs\). The contents of the node object in the SLS are slightly different
for each node type.

Refer to the OEM documentation for information about the node architecture, installation, and cabling.

For this procedure, a new object must be created in the SLS and modifications will be required to the Slingshot HSN topology.

> ## Prerequisites

Reviewer: If any scripts from the docs repo are used in the procedure, we should also add that as a prereq

Author: No scripts from the docs repo are used.

> - The Cray command line interface \(CLI\) tool is initialized and configured on the system.

Reviewer: ```suggestion
- The Cray command line interface \(CLI\) tool is initialized and configured on the system. See [Configure the Cray CLI](../configure_cray_cli.md).
```

## Procedure

1. Create a new node object in SLS.

    New node objects require the following information:
    - `Parent`: component name (xname) of the new node's BMC

>     - `Xname`: component name (xname) of the new node

Reviewer: ```suggestion
    - `Xname`: xname of the new node
```

>     - `Role`: Either `Compute` or `Application`

Reviewer: ```suggestion
    - `Role`: `Compute` or `Application`
```

>     - `Aliases`: Array of aliases for the node, for compute nodes, this is in the form of `nid0000`

Reviewer: ```suggestion
    - `Aliases`: Array of aliases for the node. For compute nodes, this is in the form of `nid0000`
```

    - NID: The Node ID integer for the node, applies only to compute nodes
    - SubRole: Such as `UAN`, `Gateway`, or other valid HSM SubRoles

>     1. If adding a compute node:

Reviewer: ```suggestion
    - If adding a compute node:
```

            }' https://api-gw-service-nmn.local/apis/sls/v1/hardware | jq
        ```

>     1. If adding a UAN:

Reviewer: ```suggestion
    - If adding a UAN:
```

            }' https://api-gw-service-nmn.local/apis/sls/v1/hardware
        ```

> 2. Create a new `MgmtSwitchConnector` object in SLS.

Reviewer: ```suggestion
1. Create a new `MgmtSwitchConnector` object in SLS.
```

>     The `MgmtSwitchConnector` connector is used by the `hms-discovery` job to determine which management switch port the node's BMC is connected to. The SLS requires the following information:

Reviewer: ```suggestion
    The `MgmtSwitchConnector` connector is used by the `hms-discovery` job to determine which management switch port is connected to the node's BMC. The SLS requires the following information:
```

    - The management switch port that the new node's BMC is connected to
    - `Xname`: The component name (xname) for the `MgmtSwitchConnector` in the form of `xXcCwWjJ`
      - `X` is the rack number
      - `C` is the chassis

>         - If the destination `LeafBMC` switch is within an standard rack this should be `0`

Reviewer: ```suggestion
        - If the destination `LeafBMC` switch is within a standard rack, then this should be `0`
```

>         - If the destination `LeafBMC` switch is within an standard rack this should be `0`.

Reviewer 2: ```suggestion
        - If the destination `LeafBMC` switch is within an standard rack this should be `0`
```

>         - If the destination `LeafBMC` switch is located within a air-cooled chassis in an EX2500 cabinet, then this should be `4`

Reviewer: ```suggestion
        - If the destination `LeafBMC` switch is located within an air-cooled chassis in an EX2500 cabinet, then this should be `4`
```

      - `W` is the rack U position of the management network leaf switch
      - `J` is the switch port number

>     - `NodeNics`: The component name (xname) of the new node's BMC. This field is an array in the payloads below, but should only contain 1 element.

Reviewer 2: ```suggestion
    - `NodeNics`: The component name (xname) of the new node's BMC; this field is an array in the payloads below, but should only contain one element
```

>       - `VendorName`: this field varies depending on the OEM for the management switch; for example, if the BMC is plugged into port 36 of the switch the following vendor names could apply:

Reviewer: ```suggestion
      - `VendorName`: This field varies depending on the OEM for the management switch; for example, if the BMC is plugged into switch port 36, then the following vendor names could apply:
```

        }' https://api-gw-service-nmn.local/apis/sls/v1/hardware | jq
    ```

### Install the Node Hardware in the Rack

1. Install the new node hardware in the rack and connect power cables, HSN cables, and management network cables \(if it has not already been installed\).

>     If the node was added before modifying the SLS, then the node's BMC should have been able to DHCP with Kea, and there will be an unknown MAC address in HSM Ethernet interfaces table.

Reviewer: ```suggestion
    If the node was added before modifying the SLS, then the node's BMC should have been able to DHCP with Kea, and there will be an unknown MAC address in the HSM Ethernet interfaces table.
```

    Refer to the OEM documentation for the node for information about the hardware installation and cabling.

### Power on and Boot Compute Node

1. Power on the node to boot the BMC.

1. Wait for the `hms-discovery` cronjob to run, and for DNS to update.

>     The `hms-discovery` cronjob will attempt to identity Node and BMC MAC addresses from the HSM Ethernet interfaces table with the connection information present in SLS to correctly identity the new node.

Reviewer: ```suggestion
    The `hms-discovery` cronjob will attempt to correctly identity the new node by comparing node and BMC MAC addresses from the HSM Ethernet interfaces table with the connection information present in SLS.
```

> 1. (`ncn-m#`) After roughly 5-10 minutes the node's BMC should be discovered by the HSM, and the node's BMC can be resolved by using its xname in DNS.

Reviewer: ```suggestion
1. (`ncn-m#`) After roughly 5-10 minutes, the node's BMC should be discovered by the HSM, and the node's BMC can be resolved by using its xname in DNS.
```

> 1. (`ncn-m#`) After roughly 5-10 minutes the node's BMC should be discovered by the HSM, and the node's BMC can be resolved by using its component name (xname) in DNS.

Reviewer 2: ```suggestion
1. (`ncn-m#`) After roughly 5-10 minutes the node's BMC should be discovered by the HSM, and the node's BMC can be resolved by using its xname in DNS.
```

    ```bash
    ping x3000c0s27b0
    ```

> 1. (`ncn-m#`) Verify that discovery has completed.

Reviewer: ```suggestion
1. (`ncn#`) Verify that discovery has completed.
```

> 1. (`ncn-m#`) Verify that discovery has completed. The

Reviewer 2: ```suggestion
1. (`ncn-m#`) Verify that discovery has completed.
```

    ```bash

>     cray hsm inventory redfishEndpoints describe x3000c0s27b0

Reviewer: ```suggestion
    cray hsm inventory redfishEndpoints describe x3000c0s27b0 --format toml
```

    ```

    Example output:

>     ```text

Reviewer: ```suggestion
    ```toml
```

    ID = "x3000c0s2b0"
    Type = "NodeBMC"

>     Hostname = ""

Author: Done

    - If the last discovery state is `HTTPsGetFailed` or `ChildVerificationFailed`, then an error has
      occurred during the discovery process.

1. (`ncn-m#`) Verify that the nodes are enabled in the HSM.

    ```bash

>     cray hsm state components describe x3000c0s27b0n0

Reviewer: ```suggestion
    cray hsm state components describe x3000c0s27b0n0 --format toml
```

    ```

    Example output:

>     ```text

Reviewer: ```suggestion
    ```toml
```

    Type = "Node"

>     Enabled = **true**

Reviewer: ```suggestion
    Enabled = true
```

    - If the last discovery state is `DiscoveryStarted` then the BMC is currently being inventoried by HSM.
    - If the last discovery state is `HTTPsGetFailed` or `ChildVerificationFailed` then an error occurred during the discovery process.

> 1. (`ncn-m#`) Enable the nodes in the HSM database \(in this example, the nodes are `x3000c0s27b1n0-n3`\).

Reviewer: ```suggestion
1. (`ncn#`) Enable the nodes in the HSM database \(in this example, the nodes are `x3000c0s27b1n0-n3`\).
```

    ```bash
    cray hsm state components bulkEnabled update --enabled true \

>     --component-ids x3000c0s27b1n0,x3000c0s27b1n1,x3000c0s27b1n2,x3000c0s27b1n3

Reviewer: ```suggestion
        --component-ids x3000c0s27b1n0,x3000c0s27b1n1,x3000c0s27b1n2,x3000c0s27b1n3
```

    ```

> 1. Verify that the correct firmware versions for node BIOS, BMC, HSN NICs, GPUs, and so on.

Reviewer: ```suggestion
1. Verify that the correct firmware versions are present for node BIOS, BMC, HSN NICs, GPUs, and so on.
```

> 1. (`ncn-m#`) If necessary, update the firmware.

Reviewer: ```suggestion
1. (`ncn#`) If necessary, update the firmware.
```

    ```bash
    cray fas actions create CUSTOM_DEVICE_PARAMETERS.json
    ```

    See [Update Firmware with FAS](../firmware/Update_Firmware_with_FAS.md).

> 1. (`ncn-m#`) Use the Boot Orchestration Service \(BOS\) to power on and boot the nodes.

Reviewer: ```suggestion
1. (`ncn#`) Use the Boot Orchestration Service \(BOS\) to power on and boot the nodes.
```

>     --operation reboot --limit x3000c0s27b0n0,x3000c0s27b0n1,x3000c0s27b0n2,x3000c0s27b00n3

Reviewer: ```suggestion
        --operation reboot --limit x3000c0s27b0n0,x3000c0s27b0n1,x3000c0s27b0n2,x3000c0s27b00n3
```

    ```

> 1. Verify the chassis status LEDs indicate normal operation.

Reviewer: ```suggestion
1. Verify that the chassis status LEDs indicate normal operation.
```
